controllers/articles.controller.js

```diff
@@ -15,10 +18,27 @@ function getArticleById(req, res, next) {
 }
 function getAllArticles(req, res, next) {
   selectAllArticles()
-    .then((articles) => {     
+    .then((articles) => {
       res.status(200).send({ articles })
     })
-    .catch(next)
+    .catch((err) => {
+      next(err)
+    })
+}
+function updateArticle(req, res, next) {
+  const { article_id } = req.params
+  const { inc_votes } = req.body
+  const promises = [updateSelectedArticle(article_id, inc_votes)]
+  if (article_id) {
+    promises.push(checkArticleExists(article_id))
```

Comment:
You don't actually need to check the article exists, as you could just throw a Promise.reject from your model if it didn't exist. However, this works fine, so up to you if you want to refactor 👍